src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1065,8 +1065,13 @@ namespace winrt::TerminalApp::implementation
     {
         auto newTabTitle = tab.Title();
 
+        // When Application window is unfocused, selected tab's FocusState also
+        // becomes Unfocused. So we have to check using FocusedTabIndex too.
+        auto focusedTabIndex = _GetFocusedTabIndex();
+
         if (_settings.GlobalSettings().ShowTitleInTitlebar() &&
-            tab.FocusState() != FocusState::Unfocused)
+            (tab.FocusState() != FocusState::Unfocused ||
```

Comment:
```
if (_settings.GlobalSettings().ShowTitleInTitlebar() && tab == _GetFocusedTab())
```
should do the job

Reply:
Done and pushed. btw `_GetFocusedTab()` can also return nullptr as per its description. Not sure its that can become a problem.